VideoWeb/VideoWeb.UnitTests/Controllers/ConferenceManagement/DismissParticipantTests.cs

```diff
@@ -137,23 +137,87 @@ public async Task should_return_accepted_when_participant_is_witness_and_judge_i
         }
 
+        [Test]
+        [TestCase(Role.QuickLinkObserver, "Observer")]
+        [TestCase(Role.QuickLinkParticipant, "Participant")]
+        public async Task should_create_an_alert_when_the_quick_link_user_is_dismissed(Role role, string expectedPrefix)
+        {
+
+            var judge = TestConference.GetJudge();
+            var participant = TestConference.Participants.First(x => x.Role == role);
+            var user = new ClaimsPrincipalBuilder()
+                .WithUsername(judge.Username)
+                .WithRole(AppRoles.JudgeRole).Build();
+            Controller = SetupControllerWithClaims(user);
+
+            string expectedBody = $"{expectedPrefix} dismissed by {judge.HearingRole}";
+
+            var result = await Controller.DismissParticipantAsync(TestConference.Id, participant.Id);
+            var typedResult = (AcceptedResult)result;
+            typedResult.Should().NotBeNull();
+
+
+            VideoApiClientMock.Verify(x => x.AddTaskAsync(TestConference.Id,
+                It.Is<AddTaskRequest>(r => r.ParticipantId == participant.Id && r.Body == expectedBody && r.TaskType == TaskType.Participant)),
+                Times.Once);
+        }
+
         [Test]
         public async Task should_create_an_alert_when_the_witness_is_dismissed()
         {
+
             var judge = TestConference.GetJudge();
-            var witness = TestConference.Participants.First(x => x.HearingRole == "Witness");
+            var participant = TestConference.Participants.First(x => x.HearingRole == "Witness");
             var user = new ClaimsPrincipalBuilder()
                 .WithUsername(judge.Username)
                 .WithRole(AppRoles.JudgeRole).Build();
+            Controller = SetupControllerWithClaims(user);
+
+            string expectedBody = $"{participant.HearingRole} dismissed by {judge.HearingRole}";
 
+            var result = await Controller.DismissParticipantAsync(TestConference.Id, participant.Id);
+            var typedResult = (AcceptedResult)result;
+            typedResult.Should().NotBeNull();
+
+
+            VideoApiClientMock.Verify(x => x.AddTaskAsync(TestConference.Id,
+                It.Is<AddTaskRequest>(r => r.ParticipantId == participant.Id && r.Body == expectedBody && r.TaskType == TaskType.Participant)),
+                Times.Once);
+        }
+
+        [Test]
+        [TestCase(Role.Judge)]
+        public async Task should_create_an_alert_with_the_correct_dismisser_role_when_the_witness_is_dismissed(Role dismisserRole)
+        {
+            string appRole;
+            switch(dismisserRole)
+            { 
+                case Role.Judge:
+                    appRole = AppRoles.JudgeRole.ToString();
+                    break;
+                case Role.StaffMember:
+                    appRole = AppRoles.StaffMember.ToString();
+                    break;
+                default:
+                    appRole = "";
+                    break;
+            }
```

Comment:
Wanted to have this as part of the test case but couldn't as they are static. I left StaffMember even though the test cases doesn't cover them just for ease when they are added back. 
Does this make sense?

Reply:
instead of using the switch cases can we not just include a second argument in the test case and pass app role along with dismisserRole, ie, like this:
      [Test]
        [TestCase(Role.Judge, "JudgeRole")]
        public async Task should_create_an_alert_with_the_correct_dismisser_role_when_the_witness_is_dismissed(Role dismisserRole, string appRole)

Reply:
It becomes a magic string issue then though. Although this switch statement feels a bit arcane itself.

Reply:
Agh. Actually I think earlier I was trying to use it like an enum instead of a string. Which makes sense. I'll have a nicer solution next commit.